### ISSUE TITLE
DefaultPostgresBinaryResolver : No postgres binaries were found, you must add an appropriate maven dependency that meets the following parameters - system: Darwin, architecture: x86_64

### ISSUE THREAD

**Issue by heatherm** [zonkyio/embedded-database-spring-test#9]
I'm on a mac (10.13.15), and have postgres 10.4 installed locally via homebrew.

I'm using Spring Boot on the application side, and when embedded-database-spring-test calls `resourceResolver.getResources(resourceLocation);` in the `DefaultPostgresBinaryResolver`, no resources are returned. The resourceLocation is set to `classpath*:postgres-darwin-x86_64.txz`.

My maven dependencies are:

```
        <dependency>
            <groupId>io.zonky.test.postgres</groupId>
            <artifactId>embedded-postgres-binaries-bom</artifactId>
            <version>10.4.0</version>
            <type>pom</type>
        </dependency>
```

I have also tried removing the embedded-postgres-binaries-bom dependency.  

Full Stack:

```
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Missing postgres binaries
	at io.zonky.test.db.flyway.DefaultFlywayDataSourceContext.lambda$reload$2(DefaultFlywayDataSourceContext.java:144) ~[embedded-database-spring-test-1.3.0.jar:na]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:529) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:599) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:617) ~[na:1.8.0_172]
	at java.util.concurrent.CompletableFuture.thenApplyAsync(CompletableFuture.java:1993) ~[na:1.8.0_172]
	at io.zonky.test.db.flyway.DefaultFlywayDataSourceContext.reload(DefaultFlywayDataSourceContext.java:133) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.postgres.FlywayEmbeddedPostgresDataSourceFactoryBean.postProcessBeforeInitialization(FlywayEmbeddedPostgresDataSourceFactoryBean.java:81) ~[embedded-database-spring-test-1.3.0.jar:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:422) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1698) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:579) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:501) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:317) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:228) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:315) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:304) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1089) ~[spring-context-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:859) ~[spring-context-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:550) ~[spring-context-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:395) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:327) ~[spring-boot-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:139) ~[spring-boot-test-2.0.2.RELEASE.jar:2.0.2.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:117) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:108) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at io.zonky.test.db.flyway.OptimizedFlywayTestExecutionListener.optimizedDbReset(OptimizedFlywayTestExecutionListener.java:135) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.flyway.OptimizedFlywayTestExecutionListener.beforeTestClass(OptimizedFlywayTestExecutionListener.java:96) ~[embedded-database-spring-test-1.3.0.jar:na]
	at org.springframework.test.context.TestContextManager.beforeTestClass(TestContextManager.java:215) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:60) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) ~[junit-4.12.jar:4.12]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190) ~[spring-test-5.0.6.RELEASE.jar:5.0.6.RELEASE]
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137) ~[junit-4.12.jar:4.12]
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68) ~[junit-rt.jar:na]
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47) ~[junit-rt.jar:na]
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242) ~[junit-rt.jar:na]
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70) ~[junit-rt.jar:na]
Caused by: java.lang.IllegalStateException: Missing postgres binaries
	at io.zonky.test.db.postgres.embedded.DefaultPostgresBinaryResolver.getPgBinary(DefaultPostgresBinaryResolver.java:68) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.EmbeddedPostgres.prepareBinaries(EmbeddedPostgres.java:701) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.EmbeddedPostgres.<init>(EmbeddedPostgres.java:119) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.EmbeddedPostgres$Builder.start(EmbeddedPostgres.java:572) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.PreparedDbProvider.createOrFindPreparer(PreparedDbProvider.java:82) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.PreparedDbProvider.<init>(PreparedDbProvider.java:63) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.shaded.com.opentable.db.postgres.embedded.PreparedDbProvider.forPreparer(PreparedDbProvider.java:56) ~[embedded-database-spring-test-1.3.0.jar:na]
	at io.zonky.test.db.flyway.DefaultFlywayDataSourceContext.lambda$reload$2(DefaultFlywayDataSourceContext.java:137) ~[embedded-database-spring-test-1.3.0.jar:na]
```

Any advice on how to debug this further would be much appreciated!

**Comment by tomix26** [comment]
Hi, thanks for your question.

You do not need to have the postgres installed locally, postgres binaries are distributed in the form of maven dependencies. In your case, the build needs the `io.zonky.test.postgres:embedded-postgres-binaries-darwin-amd64` dependency. But this dependency is part of `embedded-database-spring-test` library, so it should not be necessary to define it explicitly.

Are you sure you do not have this dependency excluded? Could you please verify that you really have this dependency in your project? If not, can you try to add the dependency explicitly?

You can also compare your maven configuration with the following sample application https://github.com/tomix26/embedded-database-demo and check that everything is set up correctly. Further, you can try to run tests of the sample project via `mvn clean test` command to rule out a configuration error on your side.

**Comment by heatherm** [comment]
Thanks for this.

The setup was near identical to the demo app. I ran tests on the demo app and they passed (Java 8). Right after, I re-ran my tests and they passed. 

Somehow, the connection to the tmp dir where the embedded-pg is now found, and pg is extracted correctly. 

I suspect that running the tests on the demo app has initialized something, but I'm not sure what. I tried deleting the `embedded-pg` directory inside `/var/folders/ys/..../T/`, but it still finds and extracts correctly, so I am unable to reproduce the error.

If I see this again I'll reopen. Thanks again for the assistance!
